Zend/zend_ast.c

```diff
@@ -1994,6 +1994,8 @@ static ZEND_COLD void zend_ast_export_ex(smart_str *str, zend_ast *ast, int prio
 			FUNC_OP("empty");
 		case ZEND_AST_ISSET:
 			FUNC_OP("isset");
+		case ZEND_AST_NAMEOF:
+			FUNC_OP("nameof");
```

Comment:
Instead of a function, wouldn't something like `::class` be better?
```php
echo $variable::name; // variable
echo ${$variable}::name; // variable
echo $object->property::name; // property
echo Object::$property::name; // property
echo Enum::Case::name; // Case
echo Object::Const::name; // Const
echo myFunction(...)::name; // myFunction
echo MY_CONSTANT::name; // MY_CONSTANT
```
etc.